### PR TITLE
add format check for services flake in dev/flake.nix

### PR DESCRIPTION
**Pull request by conscious-puppet**
adding `check` in `test/flake.nix` checks for the formatting across services flake

```
❯ nix run github:srid/nixci --option sandbox true .
```

```
treefmt-check> Initialized empty Git repository in /build/project/.git/
treefmt-check> Initialized empty Git repository in /build/project/.git/
treefmt-check> treefmt 0.6.0
treefmt-check> 0 files changed in 2ms (found 2, matched 1, cache misses 1)
treefmt-check> On branch main
treefmt-check> nothing to commit, working tree clean
treefmt-check> treefmt 0.6.0
treefmt-check> 0 files changed in 20ms (found 42, matched 26, cache misses 26)
treefmt-check> On branch main
treefmt-check> nothing to commit, working tree clean
```

### CODE REVIEW

File: test/flake.nix [hunk 1]
```diff
@@ -5,19 +5,28 @@
...
       ];
-      perSystem = { self', pkgs, system, lib, ... }: {
+      perSystem = { self', pkgs, system, lib, config, ... }: {
         _module.args.pkgs = import inputs.nixpkgs {
           inherit system;
           # Required for elastic search
           config.allowUnfree = true;
         };
+        treefmt = {
+          projectRootFile = "flake.nix";
```
Comment (shivaraj-bh) [inline test/flake.nix:24]:
```suggestion
          projectRoot = inputs.services-flake;
          projectRootFile = "flake.nix";
```

Like this you might not need to create another check

Comment (conscious-puppet) [reply test/flake.nix:24]: sure i'll give it a try

**Comment by shivaraj-bh** [comment]
Looks like it is working fine (from ubuntu-latest CI job):
Screenshot 2024-02-20 at 12 56 24 AM

File: test/flake.nix (hunk 1 above)
Comment (conscious-puppet) [reply test/flake.nix:24]: @shivaraj-bh this suggestion works. Although I had another concern. I am having to define `treefmt.programs` twice. once in `dev/flake.nix` and then again in `test/flake.nix`. Is there a way I could get the programs as well from the `${inputs.services-flake}/dev/flake.nix`

Comment (shivaraj-bh) [reply test/flake.nix:24]: Why not just add `services-flake` in inputs of `./dev`? and it should work without having to make any changes in `./test/flake.nix`

Comment (conscious-puppet) [reply test/flake.nix:24]: sure i'll try that only

Comment (shivaraj-bh) [reply test/flake.nix:24]: If this works out you can make the change in #103

Comment (conscious-puppet) [reply test/flake.nix]: adding to `services-flake` to `./dev/flake.nix` works. i've pushed that change

Comment (conscious-puppet) [reply test/flake.nix]: > If this works out you can make the change in https://github.com/juspay/services-flake/pull/103

i missed your comment before pushing to this branch. i'll close this one and push in #103 after the build is successful here

**Pull request closed without merging**